### ISSUE TITLE
Error counting the number of options in a select for rendering de options panel

### ISSUE THREAD

**Issue by didiez** [primefaces/primefaces#2029]
## 1) Environment
- PrimeFaces version: 6.0
- Does it work on the newest PrimeFaces version? No
- Application server + version: tomcat 8.5.9
- Affected browsers: All

## 2) Expected behavior
When the number of options in a select `<p:selectOneMenu>` is greater than 10, the panel showing the options is supposed to have a default height value of 200px.
The problem arises when we have groups instead of simple items (SelectItemGroup vs SelectItem).

For example: A `<p:selectOneMenu>` with 5 option groups of 7 options each, must render a container with a default fixed height of 200px.

## 3) Actual behavior
Container for the options with `height:auto` shows a huge list of options without the scroll inside de panel

## 4) Steps to reproduce
Just render the `<p:selectOneMenu>` below

## 5) Sample XHTML
```
<h:form>
    <p:selectOneMenu>
        <f:selectItems value="#{carBean.cars}"/>
    </p:selectOneMenu>
</h:form>
```
## 6) Sample bean
```
@ManagedBean
@RequestScoped
public class CarBean {
    
    public List<SelectItem> getCars(){
        int brands = 5;
        int modelsPerBrand = 7;
        List<SelectItem> cars = new ArrayList<>();
        
        for (int i = 0; i < brands; i++) {
            SelectItemGroup group = new SelectItemGroup("Brand " + (i+1));
            SelectItem[] items = new SelectItem[modelsPerBrand];
            for (int j = 0; j < modelsPerBrand; j++) {
                items[j] = new SelectItem("Model " + (i+1) + "." + (j+1));
            }
            group.setSelectItems(items);
            cars.add(group);
        }
        return cars;
    }    
}
```

**Comment by didiez** [comment]
PR with the fix #2028

**Comment by tandraschko** [comment]
Applied your PR (thanks!) + improved performance by avoiding list instances
